src/driver/rtc.c

```diff
@@ -0,0 +1,129 @@
+#include "rtc.h"
+
+#include <linux/rtc.h>
+#include <sys/ioctl.h>
+#include <sys/time.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#include <log/log.h>
+
+#include "util/file.h"
+
+/**
+ *  Constants
+ */
+static const char* RTC_FILE = "/mnt/extsd/rtc.txt";
+
+/**
+ *  Conversion Utils
+ */
+void rd2rt(const struct rtc_date* rd, struct rtc_time* rt) {
+	rt->tm_year = rd->year - 1900;
+	rt->tm_mon  = rd->month - 1;
+	rt->tm_mday = rd->day;
+	rt->tm_hour = rd->hour;
+	rt->tm_min  = rd->min;
+	rt->tm_sec  = rd->sec;
+}
+void rt2rd(const struct rtc_time* rt, struct rtc_date* rd) {
+	rd->year  = rt->tm_year + 1900;
+	rd->month = rt->tm_mon + 1;
+	rd->day   = rt->tm_mday;
+	rd->hour  = rt->tm_hour;
+	rd->min   = rt->tm_min;
+	rd->sec   = rt->tm_sec;
+}
+
+/**
+ *  Initialize RTC from file if detected,
+ *  must be formatted to ISO-8601:
+ *  YYYY-MM-DDTHH:MM:SS
+ *  2023-03-10T01:05:15
+ */
+void rtc_init() {
+	if (file_exists(RTC_FILE)) {
+		FILE* fp = fopen(RTC_FILE, "r");
+		if (fp) {
+			static char buffer[32];
+			while(fgets(buffer,sizeof(buffer),fp)) {
+				struct rtc_date rd;
+				sscanf(buffer,
+				       "%04d-%02d-%02dT%02d:%02d:%02d",
+					   &rd.year,
+					   &rd.month,
+					   &rd.day,
+					   &rd.hour,
+					   &rd.min,
+					   &rd.sec);
+			    rtc_set_clock(&rd);
+			}
+			fclose(fp);
+		}
+	}
+	
+	rtc_timestamp();
+}
+
+/**
+ *  Format RTC in a standard format
+ */
+static inline
```

Comment:
please use `clang-format` to format these files, a config file is found in the root of the repository.
integrations from common editors are available.

Reply:
I was using notepad++, I will refactor.